### PR TITLE
draft 13 and more fields in rp

### PR DESCRIPTION
**Pull request by cgeorgilakis**

### CODE REVIEW

File: design/oidc_federation.md [hunk 1]
```diff
@@ -107,8 +108,9 @@ We need to extend the current OpenID Connect 1.0 Identity provider model (`OIDCI
 - `organization_name` (optional): A human readable name representing the organization owning the RP (Keycloak).
 - `authority_hints` (required): the entity identifier(s) of intermediate entities or trust anchor(s) that Keycloak RP belongs to. This information is intended to be included in the self-signed entity statement of the RP. 
 - `expired`: Entity statement expiration time. The client registration will expire at this time. In the case of the explicit registration, Keycloak will need to periodically renew the registration (see [this](#explicit-registration-1) for details).
-- `trust_anchor_id` (required only in the case of the explicit client registration type): The value is the entity identifier of a trust anchor. This value is not supplied by the realm admin. 
-Instead, the OP will provide the value to the RP based on the trust anchor it chose when responding to an explicit client registration.
+- `trust_anchor_ids` (required): List containing the entity identifier of the trust anchors.
```
Comment (NicolasLiampotis) [inline design/oidc_federation.md]: "... containing the entity identifiers of the ..."

File: design/oidc_federation.md [hunk 2]
```diff
@@ -118,30 +120,28 @@ Moreover, `OIDCFedIdentityProviderFactory` needs to extend `AbstractIdentityProv
...
 
-The OIDC federation specification introduces the [.well-known/openid-federation](https://openid.net/specs/openid-connect-federation-1_0.html#federation_configuration) endpoint also for the RPs - which provides a JWT self-signed entity statement for the RP. 
-Thus, keycloak should have an additional endpoint under each tenant (realm), with the alias name of the idp prepended. 
-The relative path could follow the format: http(s)://host:port/{basepath}/realms/{realm-name}/{rp_alias}/.well-known/openid-federation.
-This will allow communicating with the other entities of the federation with the following entity identifier : http(s)://host:port/{basepath}/realms/{realm-name}/{rp_alias}, which eventually, uniquely identifies it within the whole federation. 
-Response content type MUST be set to application/jose.RP Metadata need to be constructed from the defined Federation OIDC Identity Provider.
+We propose to use authorization request in automatic registration which is already supported by Keycloak. Keycloak will try to interact with the Federated OP without going through a registration process.
```
Comment (NicolasLiampotis) [inline design/oidc_federation.md]: Replace with
"Automatic registration will allow Keycloak to act as an RP that can send authorization requests to an OP without first registering with the OP. We propose to perform the request by passing a request object by value as described in section 6.1 in OpenID Connect Core 1.0, which is already supported by Keycloak.

File: design/oidc_federation.md [hunk 3]
```diff
@@ -118,30 +120,28 @@ Moreover, `OIDCFedIdentityProviderFactory` needs to extend `AbstractIdentityProv
...
-Thus, keycloak should have an additional endpoint under each tenant (realm), with the alias name of the idp prepended. 
-The relative path could follow the format: http(s)://host:port/{basepath}/realms/{realm-name}/{rp_alias}/.well-known/openid-federation.
-This will allow communicating with the other entities of the federation with the following entity identifier : http(s)://host:port/{basepath}/realms/{realm-name}/{rp_alias}, which eventually, uniquely identifies it within the whole federation. 
-Response content type MUST be set to application/jose.RP Metadata need to be constructed from the defined Federation OIDC Identity Provider.
+We propose to use authorization request in automatic registration which is already supported by Keycloak. Keycloak will try to interact with the Federated OP without going through a registration process. 
+To support this, the request parameter should contain the signed entity statement of Federation OIDC Identity Provider saved on Keycloak. Entity Statement value is same as described in [Well-known OIDC federation endpoint for RP](#rp-well-know).
```
Comment (NicolasLiampotis) [inline design/oidc_federation.md]: I don't understand this. According to https://openid.net/specs/openid-connect-federation-1_0.html#rfc.section.9.1.1.1, the request parameter value "is a JWT whose Claims are the request parameters specified in Section 3.1.2 in OpenID Connect Core 1.0. The JWT MUST be signed and MAY be encrypted." Is this the same as the "signed entity statement of Federation OIDC Identity Provider"?

File: design/oidc_federation.md [hunk 4]
```diff
@@ -118,30 +120,28 @@ Moreover, `OIDCFedIdentityProviderFactory` needs to extend `AbstractIdentityProv
...
-Entity Statement value is same as described in [Well-known OIDC federation endpoint for RP](#well-known-oidc-federation-endpoint-for-rp). 
+The OIDC federation specification introduces the [.well-known/openid-federation](https://openid.net/specs/openid-connect-federation-1_0.html#federation_configuration) endpoint also for the RPs that support automatic registration - which provides a JWT self-signed entity statement for the RP. 
+Thus, keycloak should have an additional endpoint available only for OIDC Federation IdP that supports automatic registration under each tenant (realm), with the alias name of the idp prepended. 
+The relative path could follow the format: http(s)://host:port/{basepath}/realms/{realm-name}/{rp_alias}/.well-known/openid-federation - http(s)://host:port/{basepath}/realms/{realm-name}/{rp_alias} will be entity identifier, which eventually uniquely identifies RP within the whole federation. 
+This .well-known is mandatory for a successful automatic registration process. Response content type MUST be set to application/jose.RP Metadata need to be constructed from the defined Federation OIDC Identity Provider.
```
Comment (NicolasLiampotis) [inline design/oidc_federation.md]: "This .well-known endpoint is mandatory..."

File: design/oidc_federation.md [hunk 5]
```diff
@@ -118,30 +120,28 @@ Moreover, `OIDCFedIdentityProviderFactory` needs to extend `AbstractIdentityProv
...
-If the realm admin presses this button, keycloak will send a POST request with the body containing its signed entity Statement to Federation OP federation_registration_endpoint. 
-Entity Statement value is same as described in [Well-known OIDC federation endpoint for RP](#well-known-oidc-federation-endpoint-for-rp). 
+The OIDC federation specification introduces the [.well-known/openid-federation](https://openid.net/specs/openid-connect-federation-1_0.html#federation_configuration) endpoint also for the RPs that support automatic registration - which provides a JWT self-signed entity statement for the RP. 
+Thus, keycloak should have an additional endpoint available only for OIDC Federation IdP that supports automatic registration under each tenant (realm), with the alias name of the idp prepended. 
+The relative path could follow the format: http(s)://host:port/{basepath}/realms/{realm-name}/{rp_alias}/.well-known/openid-federation - http(s)://host:port/{basepath}/realms/{realm-name}/{rp_alias} will be entity identifier, which eventually uniquely identifies RP within the whole federation.
```
Comment (NicolasLiampotis) [inline design/oidc_federation.md]: is this rp_alias or idp_alias? in the previous line we mention "the alias name of the idp prepended"

**Review by NicolasLiampotis — CHANGES_REQUESTED** [review]
Minor edits and questions

File: design/oidc_federation.md (hunk 3 above)
Comment (cgeorgilakis) [reply design/oidc_federation.md]: You are right. I have changed this part.

File: design/oidc_federation.md (hunk 5 above)
Comment (cgeorgilakis) [reply design/oidc_federation.md]: idp_alias. I have confused with terminology because we are talking about implemmenting RP part. I will change it.

**Comment by cgeorgilakis** [comment]
I have implemented requested changes

**Review by NicolasLiampotis — APPROVED** [review]
LGTM

**Pull request merged**